core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -363,6 +344,10 @@ public RetryOnReplicaException(StreamInput in) throws IOException {
     }
 
     private final class AsyncReplicaAction extends AbstractRunnable {
+        /**
+         * The number of operations remaining before we can reply. See javadoc for {@link #operationComplete()} more.
+         */
+        private final AtomicInteger operationsUntilReply = new AtomicInteger(2);
```

Comment:
I'm working to remove that entirely right now. The same way we do the primary action and then start the async portion we'll do the replica action and start the async portion. I hadn't realized it was still left over when I sent the update.